docs/recipes/PWA.md

```diff
@@ -0,0 +1,339 @@
+- [App Installs](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Developer_guide/Installing)
+- [Online/Offline](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/Online_and_offline_events)
+- [Permissions](https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API)
+- [Cache](https://developer.mozilla.org/en-US/docs/Web/API/Cache)
+- [Cache Storage](https://developer.mozilla.org/en-US/docs/Web/API/CacheStorage)
+
+**Examples**
+
+* [`frank-lloyd-root` v0.0.3](frank-lloyd-root) sample module configures and enables PWA.
+
+## Setup
+
+Let's start by configuring One App to enable the service worker.
+
+**One App**
+
+First thing is we need to set `ONE_SERVICE_WORKER` before we run One App:
+
+```bash
+export ONE_SERVICE_WORKER=1 || true
+```
+
+This is set to `false` by default.
+
+**Root Module**
+
+Next step is to add it to your `appConfig` on your root module. The `pwa` key has
+a few options that we can use to turn on the service worker and create a web manifest.
+Here is a sample configuration:
+
+```jsx
+export default function RootModule({ children }) {
+  return children;
+}
+
+RootModule.childRoutes = () => [
+  <Route path="/" />,
+];
+
+RootModule.holocron = {
+  name: 'root-module',
+};
+
+if (!global.BROWSER) {
+  RootModule.appConfig = {
+    // along with other config
+    pwa: {
+      serviceWorker: true,
+      manifest: {
+        name: 'Pottery Store',
+        icons: [
+          {
+            src: 'https://drums.example.com/images/amphora-96.png',
+            sizes: '96x96',
+            type: 'image/png',
+          },
+        ],
+      },
+    },
+  };
+}
+```
+
+To read more on the `appConfig.pwa` and how you can turn off the service worker,
+[read the App Config documentation.][one-app-config]
+
+Once you've enabled the service worker and set up your manifest, you will
+load in a PWA ready One App. The service worker is installed automatically
+and the web manifest we've configured will be added to the rendered HTML.
+Under the hood, [`@americanexpress/one-service-worker`][one-service-worker]
+drives the service worker that One App is using.
+What you will get out of the box is caching for the `one-app` client, your modules,
+including language packs bundled in your package; One App also supports
+offline navigation, which means that when a user requests one of our pages when offline,
+`one-app` client and modules will both load up and it is because what we have already
+cached is given back to the browser when the network is down. Try testing what your modules
+do when offline.
+
+**Your development browser matters**
+
+After you've setup, explore around for browsers that provide tools for
+the service worker, the web manifest and the cache.
+Skip to the next paragraph when you've selected a browser.
+
+**Start**
+
+When you're ready to start, here are a few mini recipes to outfit your
+Holocron modules with PWA enhancements.
+
+## Offline Support
+
```

Comment:
what about listing the positives and negatives

Reply:
Added a list in b912ca5